pkg/downloadurl/main.go

```diff
@@ -0,0 +1,100 @@
+package downloadurl
+
+import (
+	_ "embed"
+	"fmt"
+	"sync"
+
+	"github.com/instill-ai/component/pkg/base"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"encoding/base64"
+	"io"
+	"net/http"
+
+	"github.com/gofrs/uuid"
+	"go.uber.org/zap"
+)
+
+const (
+	download = "DOWNLOAD_FROM_URL"
+)
+
+var (
+	//go:embed config/definitions.json
+	definitionsJSON []byte
+	//go:embed config/tasks.json
+	tasksJSON []byte
+	operator  base.IOperator
+	once      sync.Once
+)
+
+type Operator struct {
+	base.Operator
+}
+
+type Execution struct {
+	base.Execution
+}
+
+func Init(logger *zap.Logger) base.IOperator {
+	once.Do(func() {
+		operator = &Operator{
+			Operator: base.Operator{
+				Component: base.Component{Logger: logger},
+			},
+		}
+		err := operator.LoadOperatorDefinitions(definitionsJSON, tasksJSON)
+		if err != nil {
+
+			logger.Fatal(err.Error())
+		}
+	})
+	return operator
+}
+
+func (o *Operator) CreateExecution(defUID uuid.UUID, task string, config *structpb.Struct, logger *zap.Logger) (base.IExecution, error) {
+	e := &Execution{}
+	e.Execution = base.CreateExecutionHelper(e, o, defUID, task, config, logger)
+	return e, nil
+}
+
+type DownloadInput struct {
+	URL string `json:"url"`
+}
+
+func DownloadAsBase64(url string) (string, error) {
```

Comment:
Hi @miguel-ortiz-marin ,
Currently, you only use this function in test. I think you need to run this function in `Execute()` as well

Reply:
Hi @donch1989, you're absolutely right great catch. I've added a new commit that correctly uses the download function in the Execute and also asserts from the integration test!